helm_charts_updater/git.py

```
@@ -58,37 +143,85 @@ def push_changes(self, chart_version, app_name: str, version: str, old_version:
+            # Filter out dependency charts by checking if 'charts' appears
+            # in the relative path (dependency charts are in charts/ subdirs)
+            try:
+                rel_path = chart_path.relative_to(clone_path)
+            except ValueError:
+                continue
+
+            # Skip if 'charts' appears in path parts (indicates dependency)
+            if "charts" in rel_path.parts:
+                continue
+
+            with open(chart_path, "r", encoding="utf-8") as file:
```

Ruff (0.14.13)

Unnecessary mode argument

Remove mode argument

(UP015)